p2p/channel.go

```diff
@@ -380,17 +554,18 @@ func newBlockCrypt(privateKey PrivateKey, peerPublicKey PublicKey) (kcp.BlockCry
 	return blockCrypt, nil
 }
 
-func listenUDP(punchedConn *net.UDPConn) (*net.UDPConn, error) {
-	punchedConn.Close()
-
-	raddr := punchedConn.RemoteAddr().(*net.UDPAddr)
-	network := "udp4"
-	if raddr.IP.To4() == nil {
-		network = "udp"
+func errNetClose(err error) bool {
+	// Hack. See https://github.com/golang/go/issues/4373 which should expose net close error with 1.15.
+	if strings.Contains(err.Error(), "use of closed network connection") {
+		return true
 	}
-	conn, err := net.ListenUDP(network, punchedConn.LocalAddr().(*net.UDPAddr))
-	if err != nil {
-		return nil, fmt.Errorf("could not listen UDP: %w", err)
+	return false
+}
+
+func errPipeClosed(err error) bool {
+	// Hack. We can't check io.ErrPipeClosed as kcp wraps this error with old github.com/pkg/errors.
+	if strings.Contains(err.Error(), "io: read/write on closed pipe") {
```

Comment:
> S1008: should use 'return strings.Contains(err.Error(), "io: read/write on closed pipe")' instead of 'if strings.Contains(err.Error(), "io: read/write on closed pipe") { return true }; return false' (gosimple)